sdk/core/core-amqp/src/errors.ts

```diff
@@ -614,27 +680,24 @@ export function translate(err: AmqpError | Error): MessagingError {
 
   if (isBrowserWebsocketError(err)) {
     // Translate browser communication errors during opening handshake to generic SeviceCommunicationError
-    error = new MessagingError("Websocket connection failed.");
-    error.name = ConditionErrorNameMapper[ErrorNameConditionMapper.ServiceCommunicationError];
+    const error = new MessagingError("Websocket connection failed.");
+    error.code = ConditionErrorNameMapper[ErrorNameConditionMapper.ServiceCommunicationError];
     error.retryable = false;
     return error;
   }
 
   // instanceof checks on custom Errors doesn't work without manually setting the prototype within the error.
   // Must do a name check until the custom error is updated, and that doesn't break compatibility
   // https://github.com/Microsoft/TypeScript/wiki/Breaking-Changes#extending-built-ins-like-error-array-and-map-may-no-longer-work
-  const errorName = (err as Error).name;
+  const errorName = err.name;
   if (retryableErrors.indexOf(errorName) > -1) {
-    error.retryable = true;
-    return error;
-  }
-  if (errorName === "AbortError") {
-    error.retryable = false;
-    return error;
+    (err as any).retryable = true;
```

Comment:
Here, we are setting the `retryable` property on an error that is not `MessagingError`.
Are we now asking users to cast the errors they catch as any or `MessagingError` to determine they are retryable or not?

Reply:
I think we can actually get rid of this code block now with the changes made to err.name. This was needed when the amqp errors had the mapped condition hoisted to the error name, but now we set the error code instead (and overwrite retryable in the `isAmqpError` block).

Reply:
We do need to handle OperationTimeoutError, so I'll make some updates for that (and validate this is not an AmqpError)

Reply:
`OperationTimeoutError` is one exception to this rule. This is not an AMQP error. It originates from rhea-promise and should be treated as retryable.

See https://github.com/amqp/rhea-promise/blob/master/changelog.md#020---2019-05-17

`amqp-common` used older version of rhea-promise where `OperationTimeoutError` was indeed an AQMP error

Reply:
Found that `InsufficientCreditError` and `SendOperationFailedError` can also be thrown by rhea-promise. I now convert these to MessagingError as well, but `SendOperationFailedError` is not retryable.